src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,95 @@
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.xml.ws.soap.MTOM;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+    private EnrollmentDto enrollmentDetails;
+
+    @Mock StudentProfileService studentProfileService;
+    @Mock CourseClassService courseClassService;
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+
+    @Before
+    public void setup(){
+        enrollmentService = new EnrollmentService();
+
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+        enrollmentDetails = mock(EnrollmentDto.class);
+
+    }
+
+    @Test
+    public void processEnrollment_shouldUpdateStudentProfileWithEnrollmentDetails() throws Exception {
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+        verify(studentProfileService).updateStudentProfile(enrollmentDetails.getStudentProfile());
+    }
+
+    @Test
+    public void processEnrollment_shouldEnrollStudentClasses() throws Exception {
+
+        Student student = mock(Student.class);
+
+        Set<CourseClass> courseSet = new HashSet<>();
+        courseSet.add(mock(CourseClass.class));
+        Iterator<CourseClass> iterator = courseSet.iterator();
+
+        when(enrollmentDetails.getEnrolledClasses()).thenReturn(courseSet);
+        when(enrollmentDetails.getStudent()).thenReturn(student);
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        while (iterator.hasNext()){
+            verify(courseClassService).enrollStudentToClass(
+                    (enrollmentDetails.getStudent()),
+                    (iterator.next())
+            );
+        }
+    }
+
+    @Test
+    public void processEnrollment_shouldNotifyStudentWithEnrollmentDetails() throws Exception {
+        enrollmentService.processEnrollment(enrollmentDetails);
+        verify(enrollmentNotificationService).emailStudent(
+                ArgumentMatchers.eq(StudentEmailType.ENROLLMENT),
+                ArgumentMatchers.eq(enrollmentDetails)
+        );
+    }
+
+    @Test
+    public void processEnrollment_shouldPrintStudentsDetails() throws Exception {
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        verify(studyLoadFormatter).format(enrollmentDetails);
```

Comment:
No need to verify this one. This method should have been mocked instead and its output be passed into `systemService.print()` method as shown below:

```java
   @Test
    public void processEnrollment_shouldPrintStudentsDetails() throws Exception {
	    StudyLoad studyLoad = mock(StudyLoad.class);
	    when(studyLoadFormatter.format(enrollmentDetails)).thenReturn(studyLoad);

	    enrollmentService.processEnrollment(enrollmentDetails);

	    verify(systemService).print(studyLoad);
    }
```

As a matter of fact, calling `studyLoadFormatter.format(enrollmentDetails)` return **null** (since it was not mocked) and is passed to `systemService.print()` which is a very trivial case. Trivial cases are often bad tests.